### ISSUE TITLE
Stratis.BitcoinD bombing-out with null reference at CachedCoinView.EvictAsync()

### ISSUE THREAD

**Issue by herbepau** [stratisproject/StratisBitcoinFullNode#1858]
My Bitcoin fullnode (Stratis.BitcoinD) more than occasionally bombs-out.  I'm not doing anything in particular when this happens i.e. I'm not interacting with a wallet UI etc.

The crashing seems to occur consistently after about 10 mins of running, although that's not a definite.

This code is running from master. 

![image](https://user-images.githubusercontent.com/6311304/43711577-1e8b8a6e-996b-11e8-8a8c-58e0b1e7ed28.png)

**Comment by dangershony** [comment]
Thanks @herbepau without logs or a repro mechanism this will be hard to follow up on

**Comment by fassadlr** [comment]
Closing as there has been many improvements / optimizations around `CachedCoinView`. No issues has been reported since #3188 and #3149 also improved the fluishing mechanism.
